vignettes/connecting_to_sql.Rmd

```diff
@@ -0,0 +1,80 @@
+---
+title: "Connecting to SQL"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{connecting_to_sql}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+R can be used to execute SQL scripts to extract data from a database as well as querying the database directly via R. There are three primary ways to do this:
+
+1. executing a separate SQL script from R
+2. writing strings of SQL code in your R script
+3. using [dbplyr](https://dbplyr.tidyverse.org/) to query a database using R code
+
+Which you use will depend on how comfortable with SQL and R and also if you already have existing SQL scripts that you want to execute or you’re writing new database queries. This vignette focuses on that first example, using the `get_clean_sql()` function to read in a separate SQL script and execute from R.
+
+For more information on the other two methods, or on troubleshooting the connection between R and SQL in the Department for Education (DfE), please see the [Interacting with a SQL database section of our Analysts' Guide](https://dfe-analytical-services.github.io/analysts-guide/learning-development/r.html#interacting-with-a-sql-database-from-within-r-scripts).
+
+## Connecting to a database
+
+Usually in the DfE we use a combination of [odbc](https://odbc.r-dbi.org/) and [DBI](https://dbi.r-dbi.org/) to connect to our SQL databases.
+
+How you connect will vary depending on whether you're running R code on your laptop, or as a part of a deployed R Shiny, for running code on your laptop you can automatically use your Windows login (a trusted connection) to grant you access to the database, as the package can automatically detect your user details.
+
+```{r local connection, eval=FALSE}
+# Library calls ====
+
+library(odbc)
+library(DBI)
+
+# Database connection ====
+
+con <- DBI::dbConnect(odbc::odbc(),
+  Driver = "ODBC Driver 17 for SQL Server",
+  Server = "server_name",
+  Database = "database_name",
+  UID = "",
+  PWD = "",
+  Trusted_Connection = "Yes"
+)
+```
+
+For advice on finding your database details, or connecting to a SQL database from an R Shiny app that is deployed on a server, please contact the [Statistics Development Team](mailto:statistics.development@education.gov.uk) who will be able to advise on the setup and steps required.
+
+## Reading a SQL script into R
+
+There are a number of standard characters found in SQL scripts that can cause issues when reading in a SQL script within R, we have created the `get_clean_sql()` function to assist with this. Assume you have connected to the database and assigned that connection to a `con` variable, you would then use the following line to read a cleaned version of your SQL script into R.
```

Comment:
-> "in a SQL script within R _and_ the `get_clean_sql()` function has been created to assist with this."